toolkit/tools/pkg/imagecustomizerlib/customizeverity.go

```diff
@@ -154,34 +174,54 @@ func updateGrubConfigForVerity(rootfsVerity imagecustomizerapi.Verity, rootHash
 	return nil
 }
 
-func constructVerityKernelCmdlineArgs(rootfsVerity imagecustomizerapi.Verity, rootHash string,
+func constructVerityKernelCmdlineArgs(verityMetadata map[string]verityDeviceMetadata,
 	partIdToPartUuid map[string]string, partitions []diskutils.PartitionInfo, buildDir string,
 ) ([]string, error) {
-	// Format the dataPartitionId and hashPartitionId using the helper function.
-	formattedDataPartition, err := systemdFormatPartitionId(rootfsVerity.DataDeviceId,
-		rootfsVerity.DataDeviceMountIdType, partIdToPartUuid, partitions, buildDir)
-	if err != nil {
-		return nil, err
-	}
+	newArgs := []string{"rd.systemd.verity=1"}
+
+	for mountPath, metadata := range verityMetadata {
+		var hashArg, dataArg, optionsArg, hashKey string
+
+		switch mountPath {
+		case "/":
+			hashArg = "roothash"
+			dataArg = "systemd.verity_root_data"
+			hashKey = "systemd.verity_root_hash"
+			optionsArg = "systemd.verity_root_options"
+
+		case "/usr":
+			hashArg = "usrhash"
+			dataArg = "systemd.verity_usr_data"
+			hashKey = "systemd.verity_usr_hash"
+			optionsArg = "systemd.verity_usr_options"
+
+		default:
+			return nil, fmt.Errorf("unsupported verity mount path: %s", mountPath)
+		}
 
-	formattedHashPartition, err := systemdFormatPartitionId(rootfsVerity.HashDeviceId,
-		rootfsVerity.HashDeviceMountIdType, partIdToPartUuid, partitions, buildDir)
-	if err != nil {
-		return nil, err
-	}
+		formattedDataPartition, err := systemdFormatPartitionUuid(metadata.dataPartUuid,
+			imagecustomizerapi.MountIdentifierTypePartUuid, partitions, buildDir)
+		if err != nil {
+			return nil, err
+		}
 
-	formattedCorruptionOption, err := SystemdFormatCorruptionOption(rootfsVerity.CorruptionOption)
-	if err != nil {
-		return nil, err
-	}
+		formattedHashPartition, err := systemdFormatPartitionUuid(metadata.hashPartUuid,
+			imagecustomizerapi.MountIdentifierTypePartUuid, partitions, buildDir)
```

Comment:
You still need to use the `Verity.DataDeviceMountIdType` and `Verity,HashDeviceMountIdType` values here.

Reply:
I put them into the metadata, I checked cosi logic and it seems no impact.